pycbc/waveform/generator.py

```diff
@@ -829,6 +775,321 @@ def generate(self, **kwargs):
+            for param in kwargs if param not in self.location_args}
+        hp, hc = self.rframe_generator.generate(**rfparams)
+        if isinstance(hp, TimeSeries):
+            df = self.current_params['delta_f']
+            hp = hp.to_frequencyseries(delta_f=df)
+            hc = hc.to_frequencyseries(delta_f=df)
+            # time-domain waveforms will not be shifted so that the peak amp
+            # happens at the end of the time series (as they are for f-domain),
+            # so we add an additional shift to account for it
+            tshift = 1./df - abs(hp._epoch)
+        else:
+            tshift = 0.
+        hp._epoch = hc._epoch = self._epoch
+        h = {}
+        if self.detector_names != ['RF']:
+            for detname, det in self.detectors.items():
+                # apply the time shift
+                tc = self.current_params['tc'] + \
+                    det.time_delay_from_earth_center(self.current_params['ra'],
+                         self.current_params['dec'], self.current_params['tc'])
+                dethp = apply_fd_time_shift(hp, tc+tshift, copy=True)
+                dethc = apply_fd_time_shift(hc, tc+tshift, copy=True)
+                if self.recalib:
+                    # recalibrate with given calibration model
+                    dethp = self.recalib[detname].map_to_adjust(
+                        dethp, **self.current_params)
+                    dethc = self.recalib[detname].map_to_adjust(
+                        dethc, **self.current_params)
+                h[detname] = (dethp, dethc)
+        else:
+            # no detector response, just use the + polarization
+            if 'tc' in self.current_params:
+                hp = apply_fd_time_shift(hp, self.current_params['tc']+tshift,
+                                         copy=False)
+                hc = apply_fd_time_shift(hc, self.current_params['tc']+tshift,
+                                         copy=False)
+            h['RF'] = (hp, hc)
+        if self.gates is not None:
+            # resize all to nearest power of 2
+            hps = {}
+            hcs = {}
+            for det in h:
+                hp = h[det]
+                hc = h[det]
+                hp.resize(ceilpow2(len(hp)-1) + 1)
+                hc.resize(ceilpow2(len(hc)-1) + 1)
+                hps[det] = hp
+                hcs[det] = hc
+            hps = strain.apply_gates_to_fd(hps, self.gates)
+            hcs = strain.apply_gates_to_fd(hps, self.gates)
+            h = {det: (hps[det], hcs[det]) for det in h}
+        return h
+
+    @staticmethod
+    def select_rframe_generator(approximant):
+        """Returns a radiation frame generator class based on the approximant
+        string.
+        """
+        return select_waveform_generator(approximant)
+
+
+class FDomainDetFrameModesGenerator(BaseFDomainDetFrameGenerator):
+    """Generates frequency-domain waveform modes in a specific frame.
```

Comment:
It looks like there's a naming issue with this class (and probably the class above as well). This is *not* generating the waveform modes in a specific frame, it's just taking the waveform modes, setting an appropriate epoch and setting the 0 point of the timeseries to the coalescence time.

In other words, it's doing the conditioning needed for search filters. But that's different from coordinate transformation.

Reply:
Yeah, it's not *really* detector frame, but it does involve a detector, since you need that in order to know what time shift to apply. I know it's not perfect, but left as-is, since "DetFrame" is used in multiple places. There's really no single word that can catch all the scenarios.